File: leetcode/js/check-if-digits-are-equal-in-string-after-operations-i.js
/**
 * @param {string} s
 * @return {boolean}
 */
var hasSameDigits = function (s) {
  while (s.length > 2) {
    let temp = "";

    for (let i = 0; i < s.length - 1; i++) {
      const num = (Number(s[i]) + Number(s[i + 1])) % 10;
      temp += String(num);
    }

    s = temp;
  }

  return s[0] === s[1];
};

File: leetcode/rust/check-if-digits-are-equal-in-string-after-operations-i.rs
impl Solution {
    pub fn has_same_digits(s: String) -> bool {
        let mut s_nums: Vec<u32> = s.chars().map(|c| c.to_digit(10).unwrap()).collect();

        while s_nums.len() > 2 {
            let mut temp = Vec::new();

            for nums in s_nums.windows(2) {
                let num = (nums[0] + nums[1]) % 10;

                temp.push(num);
            }

            s_nums = temp;
        }

        s_nums[0] == s_nums[1]
    }
}
